webrender/src/clip_scroll_node.rs

```diff
@@ -377,20 +372,24 @@ impl ClipScrollNode {
             return;
         }
 
-        let combined_outer_screen_rect = match screen_outer_rect {
-            Some(outer_rect) => {
-                // If this clips outer rectangle is completely enclosed by the clip
-                // chain's inner rectangle, then the only clip that matters from this point
-                // on is this clip. We can disconnect this clip from the parent clip chain.
-                if state.combined_inner_clip_bounds.contains_rect(&outer_rect) {
-                    current_clip_chain = None;
-                }
-                outer_rect.intersection(&state.combined_outer_clip_bounds)
-                    .unwrap_or_else(DeviceIntRect::zero)
-            }
-            None => state.combined_outer_clip_bounds,
-        };
+        // All clipping ClipScrollNodes should have outer rectangles, because they never
+        // use the BorderCorner clip type and they always have at last one non-ClipOut
+        // Rectangle ClipSource.
+        let screen_outer_rect = screen_outer_rect.expect("Clipping node didn't have outer rect.");
```

Comment:
out of curiosity, is this a recommended use of shadowing local variable names, or just your preference?

Reply:
I don't know of any official recommendation around variable shadowing, but I know that it isn't totally uncommon for unwrapping Option<...> types. I felt okay using it here for that reason.